### DISCUSSION TITLE
How to use relative urls for login/consent/logout pages?

### DISCUSSION THREAD

**Discussion by hesmar** [ory/hydra#3309] [Q&A] [answered]
I would like to use hydra on an embedded device which has no fixed IP/Hostname. Therefore, it is necessary to redirect the user to the login and consent page using a relative url. Everything is behind a reverse proxy which defines the paths, so that the urls for the web browser become the following:

```
https://<dynamic_ip>/oauth      -> hydra services
https://<dynamic_ip>/ui/login   -> login page
https://<dynamic_ip>/ui/consent -> consent page
https://<dynamic_ip>/ui/logout  -> logout page
```

Looking at the configuration reference (https://www.ory.sh/docs/hydra/reference/configuration) it should be possible to set relativ paths:

```
## urls ##
#
urls:
  
  ## login ##
  #
  # Sets the OAuth2 Login Endpoint URL of the OAuth2 User Login & Consent flow. Defaults to an internal fallback URL showing an error.
  #
  # Examples:
  # - https://my-login.app/login
  # - /ui/login
  # 
  # Set this value using environment variables on
  # - Linux/macOS:
  #    $ export URLS_LOGIN=<value>
  # - Windows Command Line (CMD):
  #    > set URLS_LOGIN=<value>
  #
  login: https://my-login.app/login
```

But if I set the `urls.login` to the `/ui/login`, hydra tells me that this is not a valid URI:

```
hydra-hydra-1          | The configuration contains values or keys which are invalid:
hydra-hydra-1          | urls.login: /ui/login
hydra-hydra-1          |             ^-- "/ui/login" is not valid "uri"
```

Is it somehow possible to achieve redirection not knowing the full?

**Comment by vinckr** [comment, marked as answer]
Hello @hesmar 
Apologies for the late reply here, going through some older unanswered discussions atm. 
Ory Hydra does not currently support setting urls.login (or similar endpoints) to a relative path such as /ui/login. While the configuration reference shows examples with both absolute and relative URLs, in practice, Hydra enforces that these values must be valid URIs, which means a fully qualified URL with scheme and host.
The only workaround I can think of is to use a reverse proxy for routing, but Hydra itself still requires absolute URLs in its configuration.
Let me know if you have found another solution since.
